staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -304,6 +298,74 @@ func (w *WebhookAuthorizer) Authorize(ctx context.Context, attr authorizer.Attri
 
 }
 
+func resourceAttributesFrom(attr authorizer.Attributes) *authorizationv1.ResourceAttributes {
+	ret := &authorizationv1.ResourceAttributes{
+		Namespace:   attr.GetNamespace(),
+		Verb:        attr.GetVerb(),
+		Group:       attr.GetAPIGroup(),
+		Version:     attr.GetAPIVersion(),
+		Resource:    attr.GetResource(),
+		Subresource: attr.GetSubresource(),
+		Name:        attr.GetName(),
+	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AuthorizeWithSelectors) {
+		// If there's an error while parsing selectors, we can safely ignore the selector
+		// This is because selectors only narrow, so if broader permission is available, the request can be allowed.
+		if selector, selectorErr := attr.ParseFieldSelector(); selectorErr == nil && selector != nil {
+			retRequirements := []authorizationv1.FieldSelectorRequirement{}
+			for _, requirement := range selector.Requirements() {
+				retRequirement := authorizationv1.FieldSelectorRequirement{}
+				switch {
+				case requirement.Operator == selection.Equals || requirement.Operator == selection.DoubleEquals || requirement.Operator == selection.In:
+					retRequirement.Operator = metav1.LabelSelectorOpIn
+					retRequirement.Key = requirement.Field
+					retRequirement.Values = []string{requirement.Value}
+				case requirement.Operator == selection.NotEquals || requirement.Operator == selection.NotIn:
+					retRequirement.Operator = metav1.LabelSelectorOpIn
+					retRequirement.Key = requirement.Field
+					retRequirement.Values = []string{requirement.Value}
+				default:
+					// TODO decide about failing versus ignoring. The ignoring is safe, but not optimal
+				}
+				retRequirements = append(retRequirements, retRequirement)
+			}
+			ret.FieldSelector = &authorizationv1.FieldSelectorAttributes{
+				Requirements: retRequirements,
+			}
+		}
+
+		if selector, selectorErr := attr.ParseLabelSelector(); selectorErr == nil && selector != nil {
+			retRequirements := []metav1.LabelSelectorRequirement{}
+			requirements, _ := selector.Requirements()
+			for _, requirement := range requirements {
+				retRequirement := metav1.LabelSelectorRequirement{
+					Key:    requirement.Key(),
+					Values: requirement.Values().List(),
+				}
+				switch {
+				case requirement.Operator() == selection.Equals || requirement.Operator() == selection.DoubleEquals || requirement.Operator() == selection.In:
+					retRequirement.Operator = metav1.LabelSelectorOpIn
+				case requirement.Operator() == selection.NotEquals || requirement.Operator() == selection.NotIn:
+					retRequirement.Operator = metav1.LabelSelectorOpIn
+				case requirement.Operator() == selection.Exists:
+					retRequirement.Operator = metav1.LabelSelectorOpExists
+				case requirement.Operator() == selection.DoesNotExist:
+					retRequirement.Operator = metav1.LabelSelectorOpDoesNotExist
+				default:
+					// TODO decide about failing versus ignoring. The ignoring is safe, but not optimal
```

Comment:
same comment about omitting unknown ops producing `&authorizationv1.LabelSelectorAttributes{Requirements: []}` which will fail validation

I think omitting unknown ops is *probably* ok, but need to completely omit selector attributes if there are no requirements to propagate